pkgs/applications/networking/cluster/infra/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, buildGo118Module, fetchFromGitHub }:
+
+buildGo118Module rec {
```

Comment:
I think you can use something along these lines to build on darwin, change`buildGo118Module` to `buildGoModule` and on `pkgs/top-level/all-packages.nix`:

```
if stdenv.hostPlatform.isDarwin then buildGoModule = buildGo117Module;
```

NOTE: Did not tested but should work :P

Reply:
infra doesn't build with `go1.17` due to using a new feature provided in `go1.18`, so this will not work.